### ISSUE TITLE
Setting global font size & styles

### ISSUE THREAD

**Issue by peacechen** [archriss/react-native-render-html#21]
The docs & example show how to set individual tag styles with the `tagsStyles` prop.  Is there a way to set a global font size (as well as other styles) without having to specify individual tags?  I'd like to scale the entire HTML block responsively and a global size would help.

For example, given the HTML block with no explicit sizing:
```
<h2>Some heading</h2>
<p>First paragraph</p>
<p>Second paragraph</p>
```

**Comment by peacechen** [comment]
Passing in this tagsStyles prop
```
<HTML html={markup} tagsStyles={ {p: { fontSize: 24 }} }/>
```

Returns
```
Warning: Failed prop type: Invalid props.style key `fontSize` supplied to `View`.
```

How may the fontSize be applied to the <Text> element within the View?

**Comment by Exilz** [comment]
Hi,
not right now, but this would be an interesting feature.

We'd have to add a new prop instead of [this constant](https://github.com/archriss/react-native-render-html/blob/master/src/HTMLDefaultStyles.js#L3) so that the default styles are updated.

This should be a fairly easy task.

**Comment by Exilz** [comment]
Please copy paste the snippet you're trying to use, you might be a very specific case where the style doesn't flow properly.

**Comment by peacechen** [comment]
Here's the markup:
```
<p><b>Location</b><br>Your Town, United States of America</p>
<p><b>Duration</b><br>15 minutes</p>
<p><b>Description</b><br>Some big long-winded description...<br /><br />
More long-winded description...<br /><br />
Even more ...</p>
```

I'll work on a PR and submit it shortly.

**Comment by Exilz** [comment]
Thanks for the snippet, I found the source of the problem with `fontSize`.

Let's fix this and add this new prop in the next version.

**Comment by peacechen** [comment]
Created PR #22 which adds the `baseFontSize` prop.  The static styles in `HTMLDefaultStyles` needed to be changed to runtime capable functions to accommodate a prop.  The margins, heading sizes etc are dependent on baseFontSize.

Also fixed a bug with `<p>` styling.  htmlparser2 returns it as type `tag`, but the logic in renderRNElements() never applied the style because of the ternary `Wrapper === Text` that selected which style object to search for the tagName.

**Comment by peacechen** [comment]
This might be a related bug (or not, haven't dived into it), but the `<strong>` tag causes line breaks.

```
<p><b>Description</b><br>Some description...<br />
Item 1<strong>, </strong>
Item 2<strong>, </strong>
Item 3<strong>, </strong>
</p>
```

Which renders as:

**Description**
Some description...
Item 1
,
Item 2
,
Item 3
,

**Comment by Exilz** [comment]
I'm iterating on your PR, looks good but it needs a few tweaks and you haven't linted your code with the .eslintrc.

The problem of your line breaks does not come from the `strong` tag but from the line breaks inside your HTML code.

This isn't a desired behaviour, here's how htmlparser2 sees it :

![](https://puu.sh/xOSIh/9da9f90df2.png)

I'll work on it.

**Comment by Exilz** [comment]
@peacechen can you try the latest commit on the development branch ? Everything should be fixed.

**Comment by peacechen** [comment]
Dev branch works great!  Thanks for fixing the line breaks and related issues, and apologies for overlooking the linting.  Also appreciate how quickly you merged and refined the code.

**Comment by Exilz** [comment]
Great 😄  
I shipped this in `3.2.0`.
